### PR TITLE
Anchor upgrade to 0.27 (multiple refactors), add Redelegate instruction

### PR DESCRIPTION
**Pull request by aankor**

### CODE REVIEW

File: programs/marinade-finance/src/lib.rs [hunk 1]
```diff
@@ -719,10 +713,11 @@ impl<'info> DerefMut for UpdateActive<'info> {
 pub struct UpdateDeactivated<'info> {
     pub common: UpdateCommon<'info>,
 
+    /// CHECK: not important
```
Comment (luciotato) [inline programs/marinade-finance/src/lib.rs:716]: TODO: check how this is used. Is it VaLV7StdK6A3KfqyePbHyaBefjeibq9gakECFtXNM4m? the bot account

**Review by luciotato — APPROVED** [review]
All good. It is a proper refactor with no functionality changed

**Comment by ochaloup** [comment]
The branch `anchor-0.27` was deployed to `mainnet`
https://explorer.solana.com/tx/3T1qXqzTXjCMK7MQaAPP5y2ux5q27NkRXhYvkZ56K51oEdZ1kq8VqqYfDrbkYkoyUSaq3V4ay8yPVHie9vicUBFt
`mainnet` branch will contain the code that was deployed, commit [1bd5133](https://github.com/marinade-finance/liquid-staking-program/commit/1bd5133d3198c0af05a0952d1ca8cd0d1e19fad6)

The `main` branch will work as forward development branch.

**Comment by dinhtuananhne** [comment]
@aankor #

**Pull request merged**